flavors/vpp/vpp_flavor.go

```diff
@@ -32,29 +32,25 @@ func (f *Flavor) Inject() error {
 	}
 	f.injected = true
 
-	f.Base.Inject()
-
-	// Redis plugin
-	//f.Redis.Deps.PluginInfraDeps = *f.Base.FlavorLocal.InfraDeps("redis")
-	//f.RedisSync.Deps.PluginInfraDeps = *f.Base.FlavorLocal.InfraDeps("redis-datasync")
-	//f.RedisSync.KvPlugin = &f.Redis
-	//f.RedisSync.ResyncOrch = &f.Base.ResyncOrch
+	f.EtcdKafka.Inject()
```

Comment:
if f.FlavorXY == nil {
		f.FlavorXY = &etcd.FlavorEtcd{FlavorLocal: f.FlavorLocal}
	}
	f.FlavorXY.Inject()

Reply:
f.FlavorXY = &etcd.FlavorEtcd{FlavorLocal: f.FlavorLocal} == ambiguous reference, I have to write  &etcdkafka.FlavorEtcdKafka{FlavorLocal: f.FlavorEtcdKafka.FlavorLocal}

Reply:
No you need to create new instance of FlavorLocal (assign to anoymous composition field) and assign this instance to all three flavors (ETDCKafka, RPC, Redis)